packages/trader/src/AppV2/Utils/trade-types-utils.tsx

```diff
@@ -66,3 +77,11 @@ export const getDisplayedContractTypes = (
     Object.keys(trade_types)
         .filter(type => !getTradeTypeTabsList(contract_type).length || type === trade_type_tab)
         .sort((a, b) => getSortedIndex(a) - getSortedIndex(b));
+
+export const sortCategoriesInTradeTypeOrder = (trade_types: TContractType[], categories: TCategories[]) => {
+    return trade_types
+        .map((item: { value: string }) => {
+            return categories.find(category => category.id === item.value);
+        })
+        .filter(item => item != undefined) as TCategories[];
```

Comment:
Maybe `typeof  item !== 'undefined'` ? I read that it's safe, that direct comparison. Or at least `item => item !== undefined`

Reply:
I have removed the direct check of undefined